src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -266,6 +248,10 @@ public AzureAppConfigurationOptions UseFeatureFlags(Action<FeatureFlagOptions> c
                 {
                     // If UseFeatureFlags is called multiple times for the same key and label filters, last cache expiration time wins
                     multiKeyWatcher.CacheExpirationInterval = options.CacheExpirationInterval;
+
+                    // Move to the end, keeping precedence
+                    _multiKeyWatchers.Remove(multiKeyWatcher);
```

Comment:
Can you remove it before modifying the CacheExpirationInterval ?

Reply:
Sure, it would make more sense.